src/Avalonia.Base/Media/TextFormatting/Unicode/LineBreakEnumerator.cs

```diff
@@ -1432,10 +1434,19 @@ public BreakUnit(Codepoint codepoint, int start, int length)
             public Codepoint Codepoint { get; }
             public bool EndOfText { get; init; }
             public bool StartOfText { get; init; }
-            public LineBreakClass LineBreakClass { get; set; }
-            public bool Ignored { get; set; }
+            public LineBreakClass LineBreakClass { get; init; }
+            public bool Ignored { get; init; }
+            public bool Inherited { get; init; }
 
-            public bool Inherited { get; set; }
+            public BreakUnit WithLineBreakClass(LineBreakClass lineBreakClass)
+            {
+                return new BreakUnit(this, lineBreakClass) { Inherited = true };
+            }
```

Comment:
You don't really need to define With** methods with structs. Instead you can use "with" keyword.

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+WQIKkByAFAJSkC8AfKQB2MAO6lWNUgCE+pHABoqcnKQC+AbnqMSE0gFlu2hnQKNzpAG4BDKKWtY5nHqICWAFwAWtZb1KPNUmMLDAB2eywtM0Y1QliCQl1YawATCCEAGwBPUgBndygAVzB3CUJTc11XIVLZGgBzGHcNao8NeMqyatLqBqaWoTaOxmCAegAqUgARCDBCgFsYGut3V3S86yzc0ncsgAcYPM8IQoyU0hSYADNqw5LoezAwGD3VoXrSCC8YOwhgACsYCVPlcdp5DrlrIsdvsYEpgIVSrkYDB5tt3BBSMBDhBXmshNYMp8hKQ9lBrCVXM8AHTBXTMIzREzBYYMekcVhfcFQLjBCoWBiyPxcn7U6RRAUMajC75QamUCUxYLjUZxIA